src/transformers/sagemaker/training_args_sm.py

```diff
@@ -0,0 +1,89 @@
+# Copyright 2021 The HuggingFace Team. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import importlib.util
+from dataclasses import dataclass, field
+
+import torch
+
+from transformers.file_utils import cached_property, is_sagemaker_distributed_available
+from transformers.training_args import TrainingArguments
+from transformers.utils import logging
+
+
+logger = logging.get_logger(__name__)
+
+
+def is_smdistributed_available():
+    return importlib.util.find_spec("smdistributed") is not None
+
+
+if is_smdistributed_available():
+    import smdistributed.modelparallel.torch as smp
+
+
+@dataclass
+class SageMakerTrainingArguments(TrainingArguments):
+    mp_parameters: str = field(
```

Comment:
Where is this being set ?

Reply:
This is an argument needed to be able to launch the script via SageMaker (if it's not accepted, the script will error out). We have a class that converts dataclasses like this in an argument parser, so this line here is what will make all the example scripts (or any script using `SageMakerTrainer`) accept `--mp_parameters` from the CLI.

Unless I was launching my test scripts the wrong way?

Reply:
Aah got it. Makes sense.